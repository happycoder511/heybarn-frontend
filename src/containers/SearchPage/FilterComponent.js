import React from 'react';
import {
  BookingDateRangeFilter,
  BookingDateFromFilter,
  PriceFilter,
  KeywordFilter,
  SelectSingleFilter,
  SelectMultipleFilter,
  SelectRegionFilter,
} from '../../components';

/**
 * FilterComponent is used to map configured filter types
 * to actual filter components
 */
const FilterComponent = props => {
  const {
    idPrefix,
    filterConfig,
    urlQueryParams,
    initialValues,
    getHandleChangedValueFn,
    searchType,
    ...rest
  } = props;
    console.log("ðŸš€ | file: FilterComponent.js | line 26 | urlQueryParams", urlQueryParams);
  const { id, type, queryParamNames, label, config, listingType } = filterConfig;
  const { liveEdit, showAsPopup } = rest;

  const useHistoryPush = liveEdit || showAsPopup;
  const prefix = idPrefix || 'SearchPage';
  const componentId = `${prefix}.${id.toLowerCase()}`;
  const name = id.replace(/\s+/g, '-').toLowerCase();
  if (!listingType || !listingType.includes(searchType)) return null;
  switch (type) {
    case 'SelectSingleFilter': {
      return (
        <SelectSingleFilter
          id={componentId}
          label={label}
          queryParamNames={queryParamNames}
          initialValues={initialValues(queryParamNames)}
          onSelect={getHandleChangedValueFn(useHistoryPush)}
          {...config}
          {...rest}
        />
      );
    }
    case 'SelectRegionFilter': {
      return (
        <SelectRegionFilter
          id={componentId}
          label={label}
          name={name}
          queryParamNames={queryParamNames}
          initialValues={initialValues(queryParamNames)}
          onSubmit={getHandleChangedValueFn(useHistoryPush)}
          {...config}
          {...rest}
        />
      );
    }
    case 'SelectMultipleFilter': {
      return (
        <SelectMultipleFilter
          id={componentId}
          label={label}
          name={name}
          queryParamNames={queryParamNames}
          initialValues={initialValues(queryParamNames)}
          onSubmit={getHandleChangedValueFn(useHistoryPush)}
          {...config}
          {...rest}
        />
      );
    }
    case 'BookingDateRangeFilter': {
      return (
        <BookingDateFromFilter
          id={componentId}
          label={label}
          queryParamNames={queryParamNames}
          initialValues={initialValues(queryParamNames)}
          onSubmit={getHandleChangedValueFn(useHistoryPush)}
          {...config}
          {...rest}
        />
      );
    }
    case 'PriceFilter': {
      return (
        <PriceFilter
          id={componentId}
          label={label}
          queryParamNames={queryParamNames}
          initialValues={initialValues(queryParamNames)}
          onSubmit={getHandleChangedValueFn(useHistoryPush)}
          {...config}
          {...rest}
        />
      );
    }
    case 'KeywordFilter':
      return (
        <KeywordFilter
          id={componentId}
          label={label}
          name={name}
          queryParamNames={queryParamNames}
          initialValues={initialValues(queryParamNames)}
          onSubmit={getHandleChangedValueFn(useHistoryPush)}
          {...config}
          {...rest}
        />
      );
    default:
      return null;
  }
};

export default FilterComponent;
