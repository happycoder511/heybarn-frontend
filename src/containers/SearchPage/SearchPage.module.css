@import '../../styles/customMediaQueries.css';
@import '../../styles/propertySets.css';

/* Can be given to Topbar when a modal is opened
 * so that the Topbar is not rendered on top of it.*/
.topbarBehindModal {
  z-index: initial;
}

.topbar {
  @media (--viewportMedium) {
    position: fixed;
    top: 0;
    left: 0;
    /* We need to raise Topbar above .container */
    z-index: var(--zIndexTopbar);
  }
}

.container {
  /* Layout */
  width: 100vw;
  flex-grow: 1;
  transition: width 2s;

  /* display: flex; */

  @media (--viewportMedium) {
    position: relative;
    padding-top: var(--topbarHeightDesktop);
    min-height: calc(100vh - var(--topbarHeightDesktop));
  }
}
.mapOpenContainer {
  width: 62.5vw;
  transition: 2;
}

.mapClosed {
  @media (--viewportMedium) {
    flex-basis: 100% !important;
  }

  @media (--viewportLarge) {
    flex-basis: 100% !important;
  }
}
.searchResultContainer {
  position: relative;
  flex-grow: 1;
  display: flex;
  flex-direction: column;
  padding: 2px 0 0 0;

  @media (--viewportMedium) {
    padding: 18px 0 0 0;
    /**
     * .container is using flexbox,
     * This specifies that searchResultContainer is taking 60% from the viewport width
     */
    flex-basis: 50%;
  }

  @media (--viewportLarge) {
    flex-basis: 62.5%;
    padding: 23px 0 0 0;
  }
}

.error {
  color: var(--failColor);
  padding-left: 24px;
  padding-right: 24px;
}

.searchString {
  /* Search string should not break on white spaces - i.e. line-break should happen before. */
  white-space: nowrap;
}

.searchFiltersPrimary {
  display: none;

  @media (--viewportMedium) {
    display: flex;
    padding: 0 24px;
    margin-bottom: 21px;
  }

  @media (--viewportLarge) {
    padding: 0 36px;
  }
}

.searchFiltersMobile {
  padding: 16px 24px;

  @media (--viewportMedium) {
    display: none;
  }
}

.searchFiltersPanel {
  flex-grow: 1;
  display: flex;
  padding: 0 24px 24px 24px;

  @media (--viewportLarge) {
    padding: 0 36px 36px 36px;
  }
}

.listings {
  flex-grow: 1;
  display: flex;
  flex-direction: column;
  padding: 0;
}

.newSearchInProgress {
  opacity: 0.1;
}

.searchListingsPanel {
  flex-grow: 1;
}
.mapSwitch {
  padding: 1rem;
  position: absolute;
  top: 0;
  right: 10px;
  z-index: 1;
}
.mapPanel {
  @media (--viewportMedium) {
    /**
     * .container is using flexbox,
     * This specifies that mapPanel is taking 34% from the viewport width
     */
    flex-basis: 50%;

    /* Own layout settings */
    display: flex;
    height: 100%;
  }

  @media (--viewportLarge) {
    flex-basis: 37.5%;
  }
}

.sortBy {
  margin-right: 9px;
}

.sortByMenuLabel {
  @apply --marketplaceButtonStylesSecondary;
  @apply --marketplaceTinyFontStyles;
  font-weight: var(--fontWeightMedium);

  height: 35px;
  min-height: 35px;
  padding: 0 18px;
  margin: 0;
  border-radius: 4px;
  white-space: nowrap;
}

.mapWrapper {
  width: 100%;
  height: 100%;
}

.map {
  width: 100vw;
  height: 100vh;
  transition: all 2s ease;

  @media (--viewportMedium) {
    /* Map is fixed so that it doesn't scroll along search results */
    position: fixed;
    top: var(--topbarHeightDesktop);
    right: 0;

    /* Fixed content needs width relative to viewport */
    width: 50vw;
    height: calc(100vh - var(--topbarHeightDesktop));
  }

  @media (--viewportLarge) {
    width: 37.5vw;
  }
}
.mapOpening {
  left: -100%;
  /* width: 0.1vw; */
}
.mapOpened {
  left: revert;
  -webkit-transform: translate(0, 0);
}
.mapClosing {
}
.mapClosed {
  right: -120%;
  -webkit-transform: translate(-100%, 0);

  /* width: 0.1vw; */
}
