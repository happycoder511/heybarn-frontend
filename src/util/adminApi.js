// These helpers are calling FTW's own server-side routes
// so, they are not directly calling Marketplace API or Integration API.
// You can find these api endpoints from 'server/api/...' directory

import { types as sdkTypes, transit } from './sdkLoader';
import config from '../config';
import Decimal from 'decimal.js';

export const apiBaseUrl = () => {
  const port = process.env.REACT_APP_DEV_API_SERVER_PORT;
  const useDevApiServer = process.env.NODE_ENV === 'development' && !!port;

  // In development, the dev API server is running in a different port
  if (useDevApiServer) {
    return `http://localhost:${port}`;
  }

  // Otherwise, use the same domain and port as the frontend
  return `${window.location.origin}`;
};

// Application type handlers for JS SDK.
//
// NOTE: keep in sync with `typeHandlers` in `server/api-util/sdk.js`
export const typeHandlers = [
  // Use Decimal type instead of SDK's BigDecimal.
  {
    type: sdkTypes.BigDecimal,
    customType: Decimal,
    writer: v => new sdkTypes.BigDecimal(v.toString()),
    reader: v => new Decimal(v.value),
  },
];

const serialize = data => {
  return transit.write(data, { typeHandlers, verbose: config.sdk.transitVerbose });
};

const deserialize = str => {
  return transit.read(str, { typeHandlers });
};

const post = (path, body) => {
  const url = `${apiBaseUrl()}${path}`;
  const options = {
    method: 'POST',
    credentials: 'include',
    headers: {
      'Content-Type': 'application/transit+json',
    },
    body: serialize(body),
  };
  return window.fetch(url, options).then(res => {
    const contentTypeHeader = res.headers.get('Content-Type');
    const contentType = contentTypeHeader ? contentTypeHeader.split(';')[0] : null;

    if (res.status >= 400) {
      return res.json().then(data => {
        let e = new Error();
        e = Object.assign(e, data);

        throw e;
      });
    }
    if (contentType === 'application/transit+json') {
      return res.text().then(deserialize);
    } else if (contentType === 'application/json') {
      return res.json();
    }
    return res.text();
  });
};

// Fetch transaction line items from the local API endpoint.
//
// See `server/api/transaction-line-items.js` to see what data should
// be sent in the body.
export const updateAllListings = body => {
  return post('/api/update-all-listings', body);
};
