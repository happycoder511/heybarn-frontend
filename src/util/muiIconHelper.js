import React from 'react';
import BeachAccessIcon from '@mui/icons-material/BeachAccess';
import BeachAccessOutlinedIcon from '@mui/icons-material/BeachAccessOutlined';

import LockIcon from '@mui/icons-material/Lock';
import LockOutlinedIcon from '@mui/icons-material/LockOutlined';

import PowerIcon from '@mui/icons-material/Power';
import PowerOutlinedIcon from '@mui/icons-material/PowerOutlined';

import WaterDamageIcon from '@mui/icons-material/WaterDamage';
import WaterDamageOutlinedIcon from '@mui/icons-material/WaterDamageOutlined';

import WifiIcon from '@mui/icons-material/Wifi';
import SignalWifiStatusbarNullOutlinedIcon from '@mui/icons-material/SignalWifiStatusbarNullOutlined';

import LocalParkingIcon from '@mui/icons-material/LocalParking';

import QuestionMarkIcon from '@mui/icons-material/QuestionMark';

import WcIcon from '@mui/icons-material/Wc';

export const getIcon = (iconName, iconProps) => {
  let iconComponent;
  switch (iconName) {
    case 'beachAccess':
      return <BeachAccessIcon {...iconProps} />;
    case 'beachAccessOutlined':
      return <BeachAccessOutlinedIcon {...iconProps} />;

    case 'lock':
      return <LockIcon {...iconProps} />;
    case 'lockOutlined':
      return <LockOutlinedIcon {...iconProps} />;

    case 'power':
      return <PowerIcon {...iconProps} />;
    case 'powerOutlined':
      return <PowerOutlinedIcon {...iconProps} />;

    case 'water':
      return <WaterDamageIcon {...iconProps} />;
    case 'waterOutlined':
      return <WaterDamageOutlinedIcon {...iconProps} />;

    case 'internet':
      return <WifiIcon {...iconProps} />;
    case 'internetOutlined':
      return <SignalWifiStatusbarNullOutlinedIcon {...iconProps} />;

    case 'internet':
      return <WifiIcon {...iconProps} />;
    case 'internetOutlined':
      return <SignalWifiStatusbarNullOutlinedIcon {...iconProps} />;

    case 'wc':
      return <WcIcon {...iconProps} />;
    case 'wcOutlined':
      return <WcIcon {...iconProps} />;

    case 'parking':
      return <LocalParkingIcon {...iconProps} />;
    case 'parkingOutlined':
      return <LocalParkingIcon {...iconProps} />;

    default:
      break;
  }
};
