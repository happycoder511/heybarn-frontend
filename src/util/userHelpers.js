import queryString from 'query-string';
import { types as sdkTypes } from './sdkLoader';
import config from '../config';
const { Money } = sdkTypes;

/**
 * Title descriptor here
 *
 * @param {Object} value - params
 *
 * @return {String} Return value
 */
export const exampleFunction = value => value;

/**
 * Helper function to retrieve nested values from an object
 *
 * @param {Object} theObject - a JS object
 * @param {String} search - the string equivalent of an object key
 *
 * @return {Value} Whatever the value of the search param is or null
 */
export const getPropByName = (theObject, search) => {
  var result = null;
  if (!!theObject) {
    // eslint-disable-next-line
    for (const [key, value] of Object.entries(theObject)) {
      if (key === search) {
        result = value;
      } else if (typeof value === 'object') {
        result = getPropByName(value, search);
      }
      if (result) {
        break;
      }
    }
    return result;
  }
};
