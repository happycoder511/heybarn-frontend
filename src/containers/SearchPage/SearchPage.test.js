import React from 'react';
import { renderShallow } from '../../util/test-helpers';
import { fakeIntl } from '../../util/test-data';
import { SearchPageComponent } from './SearchPage';

const noop = () => null;

describe('SearchPageComponent', () => {
  it('matches snapshot', () => {
    const props = {
      location: { search: '' },
      history: {
        push: () => console.log('HistoryPush called'),
      },
      pagination: {
        page: 1,
        perPage: 12,
        totalItems: 22,
        totalPages: 2,
      },
      tab: 'listings',
      scrollingDisabled: false,
      searchInProgress: false,
      authInProgress: false,
      currentUserHasListings: false,
      intl: fakeIntl,
      isAuthenticated: false,
      onActivateListing: noop,
      onLogout: noop,
      onManageDisableScrolling: noop,
      onSearchMapListings: noop,
      sendVerificationEmailInProgress: false,
      onResendVerificationEmail: noop,
    };
    const tree = renderShallow(<SearchPageComponent {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
