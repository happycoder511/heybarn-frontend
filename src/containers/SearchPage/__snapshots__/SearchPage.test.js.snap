// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchPageComponent matches snapshot 1`] = `
<Page
  description="SearchPage.schemaDescription"
  schema={
    Object {
      "@context": "http://schema.org",
      "@type": "SearchResultsPage",
      "description": "SearchPage.schemaDescription",
      "mainEntity": Array [
        "{\\"@type\\":\\"ItemList\\",\\"name\\":\\"SearchPage.schemaMapSearch\\",\\"itemListOrder\\":\\"http://schema.org/ItemListOrderAscending\\",\\"itemListElement\\":[]}",
      ],
      "name": "SearchPage.schemaTitle",
    }
  }
  scrollingDisabled={false}
  title="SearchPage.schemaTitle"
>
  <withRouter(Connect(TopbarContainerComponent))
    className="topbar"
    currentPage="SearchPage"
    currentSearchParams={Object {}}
  />
  <div
    className="container"
  >
    <MainPanel
      className={null}
      filterConfig={
        Array [
          Object {
            "config": Object {
              "options": Array [
                Object {
                  "key": "northIsland",
                  "label": "North Island",
                },
                Object {
                  "key": "southIsland",
                  "label": "South Island",
                },
              ],
            },
            "group": "primary",
            "id": "locIsland",
            "label": "Location",
            "listingType": Array [
              "listing",
              "advert",
            ],
            "queryParamNames": Array [
              "pub_locIsland",
              "pub_locRegion",
              "pub_locDistrict",
            ],
            "type": "SelectRegionFilter",
          },
          Object {
            "config": Object {
              "max": 1000,
              "min": 0,
              "step": 5,
            },
            "group": "primary",
            "id": "price",
            "label": "Price",
            "listingType": Array [
              "listing",
            ],
            "queryParamNames": Array [
              "price",
            ],
            "type": "PriceFilter",
          },
          Object {
            "config": Object {},
            "group": "primary",
            "id": "dates",
            "label": "Dates",
            "listingType": Array [
              "listing",
              "advert",
            ],
            "queryParamNames": Array [
              "dates",
            ],
            "type": "BookingDateRangeFilter",
          },
          Object {
            "config": Object {
              "options": Array [
                Object {
                  "key": "storage",
                  "label": "Storage space",
                },
                Object {
                  "key": "creative",
                  "label": "Creative space",
                },
                Object {
                  "key": "work",
                  "label": "Work space",
                },
                Object {
                  "key": "event",
                  "label": "Event space",
                },
              ],
            },
            "group": "primary",
            "id": "preferredUse",
            "label": "Preferred Use",
            "listingType": Array [
              "listing",
              "advert",
            ],
            "queryParamNames": Array [
              "pub_preferredUse",
            ],
            "type": "SelectMultipleFilter",
          },
          Object {
            "config": Object {},
            "group": "primary",
            "id": "keyword",
            "label": "Keyword",
            "listingType": Array [
              "listing",
              "advert",
            ],
            "queryParamNames": Array [
              "keywords",
            ],
            "type": "KeywordFilter",
          },
          Object {
            "config": Object {
              "options": Array [
                Object {
                  "key": "advert",
                  "label": "Advert",
                },
                Object {
                  "key": "listing",
                  "label": "Listing",
                },
              ],
            },
            "group": "primary",
            "id": "listingType",
            "label": "Type",
            "listingType": Array [],
            "queryParamNames": Array [
              "pub_listingType",
            ],
            "type": "SelectSingleFilter",
          },
          Object {
            "config": Object {
              "options": Array [
                Object {
                  "key": "live",
                  "label": "Available Now Only",
                },
                Object {
                  "key": null,
                  "label": "All",
                },
              ],
            },
            "group": "primary",
            "id": "listingStatus",
            "label": "Status",
            "listingType": Array [
              "listing",
              "advert",
            ],
            "queryParamNames": Array [
              "pub_listingState",
            ],
            "type": "SelectSingleFilter",
          },
          Object {
            "config": Object {
              "customFormat": "m2",
              "max": 1000,
              "min": 1,
              "step": 1,
            },
            "group": "primary",
            "id": "sizeOfSpace",
            "label": "Size",
            "listingType": Array [
              "listing",
            ],
            "queryParamNames": Array [
              "pub_sizeOfSpace",
            ],
            "type": "SelectRangeFilter",
          },
          Object {
            "config": Object {
              "customFormat": "years",
              "max": 100,
              "min": 1,
              "step": 1,
            },
            "group": "primary",
            "id": "ageOfSpace",
            "label": "Age",
            "listingType": Array [
              "listing",
            ],
            "queryParamNames": Array [
              "pub_ageOfSpace",
            ],
            "type": "SelectRangeFilter",
          },
          Object {
            "config": Object {
              "options": Array [
                Object {
                  "key": "BayPlenty",
                  "label": "Bay of Plenty Region",
                  "parent": "northIsland",
                },
                Object {
                  "key": "HawkesBay",
                  "label": "Hawke's Bay Region",
                  "parent": "northIsland",
                },
                Object {
                  "key": "Manawatu",
                  "label": "Manawatū-Whanganui Region",
                  "parent": "northIsland",
                },
                Object {
                  "key": "Northland",
                  "label": "Northland Region",
                  "parent": "northIsland",
                },
                Object {
                  "key": "Taranaki",
                  "label": "Taranaki Region",
                  "parent": "northIsland",
                },
                Object {
                  "key": "Waikato",
                  "label": "Waikato Region",
                  "parent": "northIsland",
                },
                Object {
                  "key": "Wellington",
                  "label": "Wellington Region",
                  "parent": "northIsland",
                },
                Object {
                  "key": "Canterbury",
                  "label": "Canterbury Region",
                  "parent": "southIsland",
                },
                Object {
                  "key": "Otago",
                  "label": "Otago Region",
                  "parent": "southIsland",
                },
                Object {
                  "key": "Southland",
                  "label": "Southland Region",
                  "parent": "southIsland",
                },
                Object {
                  "key": "WestCoast",
                  "label": "West Coast Region",
                  "parent": "southIsland",
                },
              ],
            },
            "group": "primary",
            "id": "locRegion",
            "label": "Region",
            "listingType": Array [
              "listing",
              "advert",
            ],
            "queryParamNames": Array [
              "pub_locRegion",
            ],
            "type": null,
          },
          Object {
            "config": Object {
              "options": Array [
                Object {
                  "key": "kawerau",
                  "label": "Kawerau District",
                  "parent": "BayPlenty",
                },
                Object {
                  "key": "opotiki",
                  "label": "Ōpōtiki District",
                  "parent": "BayPlenty",
                },
                Object {
                  "key": "westernBayOfPlenty",
                  "label": "Western Bay of Plenty District",
                  "parent": "BayPlenty",
                },
                Object {
                  "key": "whakatane",
                  "label": "Whakatane District",
                  "parent": "BayPlenty",
                },
                Object {
                  "key": "centralHawkesBay",
                  "label": "Central Hawke's Bay District",
                  "parent": "BayPlenty",
                },
                Object {
                  "key": "hastings",
                  "label": "Hastings District",
                  "parent": "HawkesBay",
                },
                Object {
                  "key": "wairoa",
                  "label": "Wairoa District",
                  "parent": "HawkesBay",
                },
                Object {
                  "key": "horowhenua",
                  "label": "Horowhenua District",
                  "parent": "Manawatu",
                },
                Object {
                  "key": "manawatu",
                  "label": "Manawatu District",
                  "parent": "Manawatu",
                },
                Object {
                  "key": "ruapehu",
                  "label": "Ruapehu District",
                  "parent": "Manawatu",
                },
                Object {
                  "key": "whanganui",
                  "label": "Whanganui District",
                  "parent": "Manawatu",
                },
                Object {
                  "key": "farNorth",
                  "label": "Far North District",
                  "parent": "Northland",
                },
                Object {
                  "key": "kaipara",
                  "label": "Kaipara District",
                  "parent": "Northland",
                },
                Object {
                  "key": "whangarei",
                  "label": "Whangarei District",
                  "parent": "Northland",
                },
                Object {
                  "key": "newPlymouth",
                  "label": "New Plymouth District",
                  "parent": "Taranaki",
                },
                Object {
                  "key": "southTaranaki",
                  "label": "South Taranaki District",
                  "parent": "Taranaki",
                },
                Object {
                  "key": "hauraki",
                  "label": "Hauraki District",
                  "parent": "Waikato",
                },
                Object {
                  "key": "matamataPiako",
                  "label": "Matamata-Piako District",
                  "parent": "Waikato",
                },
                Object {
                  "key": "otorohanga",
                  "label": "Otorohanga District",
                  "parent": "Waikato",
                },
                Object {
                  "key": "southWaikato",
                  "label": "South Waikato District",
                  "parent": "Waikato",
                },
                Object {
                  "key": "thamesCoromandel",
                  "label": "Thames-Coromandel District",
                  "parent": "Waikato",
                },
                Object {
                  "key": "waikato",
                  "label": "Waikato District",
                  "parent": "Waikato",
                },
                Object {
                  "key": "waipa",
                  "label": "Waipa District",
                  "parent": "Waikato",
                },
                Object {
                  "key": "carterton",
                  "label": "Carterton District",
                  "parent": "Wellington",
                },
                Object {
                  "key": "kapitiCoast",
                  "label": "Kapiti Coast District",
                  "parent": "Wellington",
                },
                Object {
                  "key": "masterton",
                  "label": "Masterton District",
                  "parent": "Wellington",
                },
                Object {
                  "key": "southWairarapa",
                  "label": "South Wairarapa District",
                  "parent": "Wellington",
                },
                Object {
                  "key": "ashburton",
                  "label": "Ashburton District",
                  "parent": "Canterbury",
                },
                Object {
                  "key": "hurunui",
                  "label": "Hurunui District",
                  "parent": "Canterbury",
                },
                Object {
                  "key": "kaikoura",
                  "label": "Kaikoura District",
                  "parent": "Canterbury",
                },
                Object {
                  "key": "mackenzie",
                  "label": "Mackenzie District",
                  "parent": "Canterbury",
                },
                Object {
                  "key": "selwyn",
                  "label": "Selwyn District",
                  "parent": "Canterbury",
                },
                Object {
                  "key": "timaru",
                  "label": "Timaru District",
                  "parent": "Canterbury",
                },
                Object {
                  "key": "waimakariri",
                  "label": "Waimakariri District",
                  "parent": "Canterbury",
                },
                Object {
                  "key": "waimate",
                  "label": "Waimate District",
                  "parent": "Canterbury",
                },
                Object {
                  "key": "centralOtago",
                  "label": "Central Otago District",
                  "parent": "Otago",
                },
                Object {
                  "key": "clutha",
                  "label": "Clutha District",
                  "parent": "Otago",
                },
                Object {
                  "key": "queenstownLakes",
                  "label": "Queenstown-Lakes District",
                  "parent": "Otago",
                },
                Object {
                  "key": "gore",
                  "label": "Gore District",
                  "parent": "Southland",
                },
                Object {
                  "key": "southland",
                  "label": "Southland District",
                  "parent": "Southland",
                },
                Object {
                  "key": "buller",
                  "label": "Buller District",
                  "parent": "WestCoast",
                },
                Object {
                  "key": "grey",
                  "label": "Grey District",
                  "parent": "WestCoast",
                },
                Object {
                  "key": "westland",
                  "label": "Westland District",
                  "parent": "WestCoast",
                },
              ],
            },
            "group": "primary",
            "id": "locDistrict",
            "label": "District",
            "listingType": Array [
              "listing",
              "advert",
            ],
            "queryParamNames": Array [
              "pub_locDistrict",
            ],
            "type": null,
          },
          Object {
            "config": Object {
              "options": Array [
                Object {
                  "checkedIcon": "beachAccess",
                  "key": "weatherproof",
                  "label": "Weatherproof",
                  "unCheckedIcon": "beachAccessOutlined",
                },
                Object {
                  "checkedIcon": "lock",
                  "key": "secure",
                  "label": "Secure",
                  "unCheckedIcon": "lockOutlined",
                },
                Object {
                  "checkedIcon": "power",
                  "key": "mainsPower",
                  "label": "Mains power",
                  "unCheckedIcon": "powerOutlined",
                },
                Object {
                  "checkedIcon": "water",
                  "key": "waterSupply",
                  "label": "Water supply",
                  "unCheckedIcon": "waterOutlined",
                },
                Object {
                  "checkedIcon": "internet",
                  "key": "internet",
                  "label": "Internet",
                  "unCheckedIcon": "internetOutlined",
                },
                Object {
                  "checkedIcon": "wc",
                  "key": "wc",
                  "label": "Toilet",
                  "unCheckedIcon": "wcOutlined",
                },
                Object {
                  "checkedIcon": "parking",
                  "key": "parking",
                  "label": "Parking",
                  "unCheckedIcon": "parkingOutlined",
                },
              ],
              "searchMode": "has_all",
            },
            "group": "secondary",
            "id": "amenities",
            "label": "Features",
            "listingType": Array [
              "listing",
            ],
            "queryParamNames": Array [
              "pub_amenities",
            ],
            "type": "SelectMultipleFilter",
          },
          Object {
            "config": Object {
              "options": Array [
                Object {
                  "key": "anytimeUnrestricted",
                  "label": "Anytime, access is not restricted",
                },
                Object {
                  "key": "anytimeArranged",
                  "label": "Anytime, but by arrangement",
                },
                Object {
                  "key": "weekendsOnly",
                  "label": "Prefer only weekends",
                },
                Object {
                  "key": "weekdaysOnly",
                  "label": "Prefer only office hours",
                },
                Object {
                  "key": "negotiable",
                  "label": "Negotiable",
                },
              ],
              "searchMode": "has_any",
            },
            "group": "secondary",
            "id": "listingAccessFrequency",
            "label": "How much access would you permit?",
            "listingType": Array [
              "listing",
            ],
            "queryParamNames": Array [
              "pub_accessFrequency",
            ],
            "type": "none",
          },
          Object {
            "config": Object {
              "options": Array [
                Object {
                  "key": "anytimeUnrestricted",
                  "label": "All the time",
                },
                Object {
                  "key": "anytimeArranged",
                  "label": "Anytime, but I'll let you know",
                },
                Object {
                  "key": "weekdaysOnly",
                  "label": "Typically only weekdays",
                },
                Object {
                  "key": "weekendsOnly",
                  "label": "Typically only weekends",
                },
                Object {
                  "key": "infrequently",
                  "label": "Pretty infrequently",
                },
                Object {
                  "key": "dropAndCollect",
                  "label": "Once to drop off, once to collect",
                },
                Object {
                  "key": "negotiable",
                  "label": "Negotiable",
                },
              ],
              "searchMode": "has_any",
            },
            "group": "secondary",
            "id": "advertAccessFrequency",
            "label": "Access Frequency",
            "listingType": Array [
              "advert",
            ],
            "queryParamNames": Array [
              "pub_accessFrequency",
            ],
            "type": "none",
          },
          Object {
            "config": Object {
              "options": Array [
                Object {
                  "key": "noSmoking",
                  "label": "No smoking",
                },
                Object {
                  "key": "noPets",
                  "label": "No pets",
                },
                Object {
                  "key": "noGuests",
                  "label": "No guests",
                },
                Object {
                  "key": "noSignage",
                  "label": "No signage",
                },
              ],
              "searchMode": "has_any",
            },
            "group": "secondary",
            "id": "groundRules",
            "label": "Ground rules",
            "listingType": Array [
              "listing",
            ],
            "queryParamNames": Array [
              "pub_accessFrequency",
            ],
            "type": "none",
          },
        ]
      }
      history={
        Object {
          "push": [Function],
        }
      }
      isMapOpen={false}
      listings={Array []}
      mapSwitch={
        <ForwardRef(FormControlLabel)
          className="mapSwitch"
          control={
            <ForwardRef(Switch)
              checked={false}
              inputProps={
                Object {
                  "aria-label": "Show map",
                }
              }
              onChange={[Function]}
            />
          }
          label="Show map"
          labelPlacement="start"
        />
      }
      onActivateListing={[Function]}
      onCloseModal={[Function]}
      onManageDisableScrolling={[Function]}
      onMapIconClick={[Function]}
      onOpenModal={[Function]}
      pagination={
        Object {
          "page": 1,
          "perPage": 12,
          "totalItems": 22,
          "totalPages": 2,
        }
      }
      resultsCount={0}
      rootClassName={null}
      searchInProgress={false}
      searchListingsError={null}
      searchParamsAreInSync={true}
      searchParamsForPagination={Object {}}
      searchType="listing"
      showAsModalMaxWidth={768}
      sortConfig={
        Object {
          "active": true,
          "conflictingFilters": Array [
            "keyword",
          ],
          "options": Array [
            Object {
              "key": "createdAt",
              "label": "Newest",
            },
            Object {
              "key": "-createdAt",
              "label": "Oldest",
            },
            Object {
              "key": "-price",
              "label": "Lowest price",
            },
            Object {
              "key": "price",
              "label": "Highest price",
            },
            Object {
              "key": "relevance",
              "label": "Relevance",
              "longLabel": "Relevance (Keyword search)",
            },
          ],
          "queryParamName": "sort",
          "relevanceKey": "relevance",
        }
      }
      urlQueryParams={Object {}}
    />
    <withViewport(ModalInMobileComponent)
      className="mapPanel"
      id="SearchPage.map"
      isModalOpenOnMobile={false}
      onClose={[Function]}
      onManageDisableScrolling={[Function]}
      showAsModalMaxWidth={768}
    >
      <div
        className="mapWrapper"
      >
        <withRouter(SearchMapComponent)
          activeListingId={null}
          isSearchMapOpenOnMobile={false}
          listings={Array []}
          location={
            Object {
              "search": "",
            }
          }
          mapClass="mapClosed"
          onCloseAsModal={[Function]}
          onMapMoveEnd={[Function]}
          reusableContainerClassName="map"
        />
      </div>
    </withViewport(ModalInMobileComponent)>
  </div>
</Page>
`;
